src/mds/SnapRealm.h

```diff
@@ -92,7 +92,7 @@ struct SnapRealm {
     return (p != s.end() && *p <= last);
   }
 
-  inodeno_t get_subvolume_ino() {
+  inodeno_t get_subvolume_ino() const {
```

Comment:
Is this required for the debug change?

Reply:
yes, its required
or
we need to do away with the `const` specifier on the SnapRealm parameter in the dump `operator<<(...)` function

Reply:
So, `cached_subvolume_ino` is modified in `check_cache()` and that variable is marked mutable. You could just use `cached_subvolume_ino` in `operator<<`, but I guess this is fine too :)

Reply:
yeah ... also `cached_subvolume_ino` is kept `private`